src/base/bittorrent/torrentimpl.cpp

```diff
@@ -243,14 +243,14 @@ namespace
 
     namespace LT
     {
-        constexpr lt::download_priority_t toNative(const DownloadPriority priority)
+        constexpr lt::download_priority_t toNative(const DownloadPriority priority) noexcept
```

Comment:
>Have you checked that the conversion has no narrowing? otherwise it is undefined behavior 

Actually it has. But there is no undefined behavior (since integral conversions are always defined, even narrowing ones, although they are implementation-defined until c++20). But here we are dealing with values that obviously fit into the target range, so there is no problem.
>and not `noexcept`.

Then it would not have successfully compiled.
>Also applies to other new `noexcept` additions.

If you mean `toLTUnderlyingType()` then it is definitely `noexcept`. Actually it contains only call to conversion operator that is just a getter of class field that stores value of underlying type.

Seems you just dislike `noexcept` specified.
I try to apply it only in those trivial cases that definitely should not produce any exceptions.

Reply:
>>and not noexcept.
>
>Then it would not have successfully compiled.

AFAIK the compiler won't check the statements are "noexcept", `noexcept` is coder's responsibility to make sure all function/statements upholds it. For example the following won't produce errors/warnings:
```c++
void f1()
{
    throw std::runtime_error("");
}
void f2() noexcept
{
    f1();
}
```


>Seems you just dislike noexcept specified.

I don't really object it but I think it is just to cumbersome to check everything is noexcept that I don't like to bother with it.

Reply:
>But there is no undefined behavior (since integral conversions are always defined, even narrowing ones, although they are implementation-defined until c++20). 

OK. I hope "implementation-defined" don't rely on exceptions.

>But here we are dealing with values that obviously fit into the target range

Not really. Just to clarify, `DownloadPriority` underlying is `int` and `lt::download_priority_t` underlying is `std::uint8_t` and the following code is valid (I hope it is unlikely to happen):
```c++
void f1()
{
    auto a = BitTorrent::DownloadPriority(10000);
    auto b = LT::toNative(a);  // 10000 will be truncated to fit into std::uint8_t
}
```

Reply:
>>But here we are dealing with values that obviously fit into the target range
>
>Not really. 

I don't mean all possible values but all declared/used values of DownloadPriority.

I hope you are not going to seriously engage in handling obviously invalid values like `DownloadPriority(100000)`.

Reply:
>I hope you are not going to seriously engage in handling obviously invalid values like DownloadPriority(100000)

If we have sanitized the inputs at the boundaries (such as webapi) then we can ignore this problem at the core.